File: frontend/services/auth.service.ts
const BASE_URL = process.env.NEXT_PUBLIC_API_URL || "http://localhost:3000";

export interface UserResponse {
  fullName: string;
  email: string;
  role: string;
}

export interface AuthResponse {
  message: string;
  access_token: string;
  user: UserResponse;
}

export const loginUser = async (email: string, password: string): Promise<AuthResponse> => {
  const response = await fetch(`${BASE_URL}/auth/login`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      email: email.toLowerCase().trim(),
      password
    }),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message || "Login failed");
  }

  return response.json();
};

export const registerUser = async (fullName: string, email: string, password: string): Promise<AuthResponse> => {
  const response = await fetch(`${BASE_URL}/auth/register`, {
    method: "POST",
    headers: { "Content-Type": "application/json" },
    body: JSON.stringify({
      fullName,
      email: email.toLowerCase().trim(),
      password
    }),
  });

  if (!response.ok) {
    const error = await response.json();
    throw new Error(error.message || "Registration failed");
  }

  return response.json();
};

File: mobile/lib/services/auth_service.dart
import 'dart:convert';
import 'package:http/http.dart' as http;
import 'package:flutter_dotenv/flutter_dotenv.dart';

class AuthService {

  static String get baseUrl =>
      dotenv.env['BASE_URL']!;

  static Future<dynamic> login(
    String email,
    String password,
  ) async {

    final response =
        await http.post(
      Uri.parse(
        "$baseUrl/auth/login",
      ),
      headers: {
        "Content-Type":
            "application/json",
      },
      body: jsonEncode({
        "email": email,
        "password": password,
      }),
    );

    return jsonDecode(
      response.body,
    );
  }

  static Future<dynamic> register(
    String fullName,
    String email,
    String password,
  ) async {

    final response =
        await http.post(
      Uri.parse(
        "$baseUrl/auth/register",
      ),
      headers: {
        "Content-Type":
            "application/json",
      },
      body: jsonEncode({
        "fullName": fullName,
        "email": email,
        "password": password,
      }),
    );

    return jsonDecode(
      response.body,
    );
  }
}